src/parse.c

```diff
@@ -3116,11 +3120,12 @@ process_document(NetplanParser* npp, GError** error)
         g_hash_table_iter_next (&iter, &key, &value);
         missing = (NetplanMissingNode*) value;
 
-        return yaml_error(npp, missing->node, error, "%s: interface '%s' is not defined",
-                          missing->netdef_id,
-                          key);
+        ret = yaml_error(npp, missing->node, error, "%s: interface '%s' is not defined",
+                         missing->netdef_id, key);
+        goto cleanup;
```

Comment:
this goto is not needed (but doesn't hurt) as the last lines will be executed anyway

Reply:
also, this hashtable is released by netplan_parser_reset(), but that's fine keeping it here.

Reply:
Thanks for your comments!
* IMO it is nice to clear the hashtable close to its creation.. doesn't hurt to "clear" it again in netplan_parser_reset * ACK. I was pondering about adding the goto or nor. But decided to add it, as this way we don't need to think about it in the future should we add a block of code in between that if condition and the return statement